addons/pos_sale/models/pos_order.py

```diff
@@ -60,7 +60,18 @@ def sync_from_ui(self, orders):
                         self.env['sale.advance.payment.inv']._prepare_down_payment_section_values(sale_order_origin)
                     )
                 order_reference = line.name
-                sale_order_line_description = _("Down payment (ref: %(order_reference)s on \n %(date)s)", order_reference=order_reference, date=line.order_id.date_order.strftime('%m-%d-%y'))
+
+                def _get_lang():
+                    if order.partner_id.lang and not order.partner_id.is_public:
```

Comment:
Why do we need to check if the user is not public user ?

Reply:
I have copied this code mainly from `sale_order_line.py` [here](https://github.com/odoo/odoo/blob/35e933e02e26ddde85d5ea585ddbdcff2df8dc8a/addons/sale/models/sale_order_line.py#L345). We are calling the `_get_lang` [function](https://github.com/odoo/odoo/blob/5e50fab3e3aee1262fb0697410402f99e484059a/addons/sale/models/sale_order.py#L2025) in `sale_order.py` that checks for not public user. 

I thought those 2 places are closely related, since both are used to format the date on the order line, so I have duplicated this code.

Reply:
I can do more investigation if needed :)

Reply:
I think they do this because it can be created through the website and the website user is not always connected. But in our case it will always be created with the user connected on the PoS. So IMO this is not required.

Could you also add a small test ? Thanks !

Reply:
@robinengels I have added the test, could you please let me know if it's good?